corehq/apps/es/migrations/0016_add_new_index_for_cc_perf.py

```diff
@@ -0,0 +1,40 @@
+from django.conf import settings
+from django.db import migrations
+
+from corehq.apps.es.mappings.case_search_mapping import CASE_SEARCH_MAPPING
+from corehq.apps.es.migration_operations import (
+    CreateIndex,
+    DeleteOnlyIfIndexExists,
+)
+from corehq.apps.es.utils import index_runtime_name
+
+
+def _create_cc_perf_index_on_saas_envs(apps, schema_editor):
+    if settings.IS_SAAS_ENVIRONMENT:
```

Comment:
This should probably use the new setting (`ENABLE_BHA_CASE_SEARCH_ADAPTER`) or something similar, but `IS_SAAS_ENVIRONMENT` is almost certainly not correct.